keylime/common/algorithms.py

```diff
@@ -12,47 +12,30 @@ def is_accepted(algorithm: str, accepted: List[Any]) -> bool:
     return algorithm in accepted
 
 
-def _hashit(algorithm: str, data: bytes) -> bytes:
-    if algorithm == "sha1":
-        return hashlib.sha1(data).digest()
-    if algorithm == "sha256":
-        return hashlib.sha256(data).digest()
-    if algorithm == "sha384":
-        return hashlib.sha384(data).digest()
-    if algorithm == "sha512":
-        return hashlib.sha512(data).digest()
-    if algorithm == "sm3_256":
-        # SM3 is not guaranteed to be there
-        return hashlib.new("sm3", data).digest()
-
-    raise ValueError(f"Unsupported hash algorithm {algorithm}")
-
-
 class Hash(str, enum.Enum):
     SHA1 = "sha1"
     SHA256 = "sha256"
     SHA384 = "sha384"
     SHA512 = "sha512"
-    SM3_256 = "sm3_256"
+    SM3_256 = "sm3"
```

Comment:
At least now it's the same as what the Linux kernel uses: https://elixir.bootlin.com/linux/latest/source/crypto/hash_info.c#L29

I wonder whether this change now will make a difference for anyone previously having used Keylime with `sm3_256` string "somewhere" in the config.

Reply:
Right. A `grep` did not show anything in the codebase, and if it comes from a configuration file it will trigger now the ValueError exception (and the fix should be directy)

I was inclined to remove this `sm3` algorithm, or extend it with `sm4` too.

Reply:
The name `sm3_256` is also used by tpm2_tools and we pass the string value to it, so we cannot just change it to `sm3`

Reply:
Worth writing this into the file...

Reply:
Reverted the name and add some documentation